extensions/sql-migration/src/wizard/databaseBackupPage.ts

```diff
@@ -525,89 +525,105 @@ export class DatabaseBackupPage extends MigrationWizardPage {
 	}
 
 	private async getSubscriptionValues(): Promise<void> {
-		this._networkShareContainerSubscriptionDropdown.loading = true;
-		this._fileShareSubscriptionDropdown.loading = true;
-		this._blobContainerSubscriptionDropdown.loading = true;
-		try {
-			this._fileShareSubscriptionDropdown.values = await this.migrationStateModel.getSubscriptionsDropdownValues();
-			this._networkShareContainerSubscriptionDropdown.values = await this.migrationStateModel.getSubscriptionsDropdownValues();
-			this._blobContainerSubscriptionDropdown.values = await this.migrationStateModel.getSubscriptionsDropdownValues();
-			this.migrationStateModel.databaseBackup.subscription = this.migrationStateModel.getSubscription(0);
-		} catch (error) {
-			this.migrationStateModel._storageAccounts = undefined!;
-		} finally {
-			await this.loadNetworkShareStorageDropdown();
-			await this.loadFileShareStorageDropdown();
-			await this.loadblobStorageDropdown();
-			this._networkShareContainerSubscriptionDropdown.loading = false;
-			this._fileShareSubscriptionDropdown.loading = false;
-			this._blobContainerSubscriptionDropdown.loading = false;
+		if (!this.migrationStateModel.databaseBackup.subscription) {
+			this._networkShareContainerSubscriptionDropdown.loading = true;
+			this._fileShareSubscriptionDropdown.loading = true;
+			this._blobContainerSubscriptionDropdown.loading = true;
+			try {
+				this._fileShareSubscriptionDropdown.values = await this.migrationStateModel.getSubscriptionsDropdownValues();
```

Comment:
You're calling getSubscriptionDropdownValues multiple times - is that necessary? Seems like you should be able to just call it once and reuse those values.